### DISCUSSION TITLE
Monai Integration within Lightning Flash

### DISCUSSION THREAD

**Discussion by tchaton** [Project-MONAI/MONAI#2880] [Ideas]
Hey everyone,

First of all, I want to share I am really impressed by the work put into this library. Looks pretty awesome !

I am a member of the PyTorch Lightning Team and working on Lightning Flash repo: https://github.com/PyTorchLightning/lightning-flash

Lightning Flash is a collection of production ready tasks for prototyping, baselining, finetuning and solving problems with deep learning.

We currently supports 15+ tasks while integrating multiple frameworks as backends and using Lightning as the training framework.

We believe MONAI would be a very exciting integration within Flash and I wondered if anyone from MONAI would be interested to contribute 3D tasks ?

Best,
T.C

**Comment by Nic-Ma** [comment]
Marked for discussion.

Thanks.

**Comment by Nic-Ma** [comment]
Hi @tchaton ,

Thanks for the sharing, glad to see the progress of Lightning.
We already built some Lightning + MONAI tutorials:
https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/spleen_segmentation_3d_lightning.ipynb
https://github.com/Project-MONAI/tutorials/blob/master/3d_segmentation/unetr_btcv_segmentation_3d_lightning.ipynb
Maybe that's what you expected?

Thanks.

**Comment by tchaton** [comment]
Hey @Nic-Ma,

I believed we could create a 3DImageSegmentation Task in Flash and use Monai as a backend with pre-trained models, etc...

Best,
T.C

**Reply by Nic-Ma** [reply]
Hi @tchaton ,

Are the above 2 Lightning + MONAI tutorials what you expected?
If so, I think feel free to convert to some backend tasks as you want.
@wyli @ericspod What do you guys think?

Thanks.

**Reply by ericspod** [reply]
The speen tutorial would be the easiest to adapt I'd suggest, it doesn't involve any special training process other than our transform pipeline.
